source/JsonValidation.cpp

```diff
@@ -105,6 +105,18 @@ int JsonValidation::integer(
   return integer.asInt();
 }
 
+std::optional<int> JsonValidation::optional_integer(
+    const Json::Value& value,
+    const std::string& field) {
+  validate_object(
+      value, fmt::format("non-null object with integer field `{}`", field));
+  const auto& integer = value[field];
+  if (integer.isNull() || !integer.isInt()) {
+    return std::nullopt;
+  }
```

Comment:
I think we should throw an exception if field is not null and is not an integer (for instance if this is a string). We shouldn't just return `std::nullopt`, that would mean we silently ignore stuff like `"start_idx": "foo"`